### DISCUSSION TITLE
Support for field label in ADT?

### DISCUSSION THREAD

**Discussion by bohonghuang** [coalton-lang/coalton#670] [Ideas and Proposals]
Field label in Haskell helps select fields by name in pattern matching:
``` Haskell
data Person = Person { age :: Int, name :: String }
lowerCaseName :: Person -> String
lowerCaseName (Person { name = x }) = map toLower x
``` 
Any plan to support this feature?

**Comment by stylewarning** [comment]
No plans to support or not support either way.

FWIW, you can write a simple macro to allow field names if you please for your code base!

(EDIT: But it wouldn't work with pattern matching/name-based destructuring.)

**Reply by ornamentist** [reply]
Would a separate `Record` macro make more sense in Coalton than trying to change algebraic data types?
